dm/checker/checker.go

```diff
@@ -119,75 +122,51 @@ func (c *Checker) Init(ctx context.Context) (err error) {
 	rollbackHolder.Add(fr.FuncRollback{Name: "close-DBs", Fn: c.closeDBs})
 
 	c.tctx = tcontext.NewContext(ctx, log.With(zap.String("unit", "task check")))
+
+	// prepare source target do dbs concurrently
+	eg, ctx2 := errgroup.WithContext(ctx)
+	// sourceID => source targetDB mapping, only writing need used mu to avoid concurrent write map
+	var mu sync.Mutex
+	sourceTargetM := make(map[string]map[string][]*filter.Table)
+	for idx := range c.instances {
+		i := idx
```

Comment:
Not suggestion. I'm just curious about why idx has to be copied before being used? It seems idx is an integer, which should be copied instead of being moved. (I'm not familiar with golang, would there be any problems if we use the idx directly in the closure by catching it from the outside space? Or idx is caught as ref by default?)

Reply:
this is strange in go without passing copy/val, then the value of val will be whatever val is when the goroutines actually runs see more here https://riptutorial.com/go/example/3897/using-closures-with-goroutines-in-a-loop